src/main/java/be/minelabs/entity/CorrosiveEntity.java

```diff
@@ -0,0 +1,72 @@
+package be.minelabs.entity;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.GameRules;
+import net.minecraft.world.World;
+
+public class CorrosiveEntity extends Entity {
+
+    // Hold a reference to a blockpos
+    private BlockPos blockPos;
+    private int ticks = 0;
+    private double distance = 0;
+
+    public CorrosiveEntity(EntityType<? extends Entity> type, World world) {
+        super(type, world);
+    }
+
+    private CorrosiveEntity(World world, BlockPos blockPos, double distance) {
+        this(Entities.CORROSIVE_ENTITY, world);
+        this.blockPos = blockPos;
+        this.distance = distance;
+    }
+
+    public static CorrosiveEntity create(World world, BlockPos blockPos, double distance) {
+        return new CorrosiveEntity(world, blockPos, distance);
+    }
+
+
+    // On every tick, set the blockbreakinginfo of the block at the blockpos to 5
+    @Override
+    public void tick() {
+        super.tick();
+        // If the block at the blockpos has turned into air, remove the entity
+        if (this.blockPos != null && this.world.getBlockState(blockPos).isOf(net.minecraft.block.Blocks.AIR)) {
+            this.world.setBlockBreakingInfo(this.getId(), this.blockPos, 0);
+            this.remove(RemovalReason.DISCARDED);
+        }
+        if (this.blockPos != null) {
```

Comment:
I think you only want the tick function to run on the server. This check is used as a proxy because the value is probably not set on the client. Could you test this and make it more clear by checking for server world instead.

Reply:
When looking into the method `setBlockBreakingInfo()`
The Server world only has it defined as an abstract method.
And Client actually does something with it.

So if anything we could check for client and only tick on client. But since server doesn't handle it don't see the issue.

Reply:
Did you check `ServerWorld` too? It seems like something the server should coordinate. For example when one player mines something, don't other players see the progress as well?

Whether we need the server to coordinate this is up for debate. As long as all clients have the entities as well, I guess it will also render just fine, as long as the actual block breaking logic remains server side.

Reply:
`ServerWorld` only triggers the networking.
```java
public void setBlockBreakingInfo(int entityId, BlockPos pos, int progress) {
        for (ServerPlayerEntity serverPlayerEntity : this.server.getPlayerManager().getPlayerList()) {
            double f;
            double e;
            double d;
            if (serverPlayerEntity == null || serverPlayerEntity.world != this || serverPlayerEntity.getId() == entityId || !((d = (double)pos.getX() - serverPlayerEntity.getX()) * d + (e = (double)pos.getY() - serverPlayerEntity.getY()) * e + (f = (double)pos.getZ() - serverPlayerEntity.getZ()) * f < 1024.0)) continue;
            serverPlayerEntity.networkHandler.sendPacket(new BlockBreakingProgressS2CPacket(entityId, pos, progress));
        }
    } 
````
   
But `ClientWorld` effects the visual.
```java
public void setBlockBreakingInfo(int entityId, BlockPos pos, int progress) {
        this.worldRenderer.setBlockBreakingInfo(entityId, pos, progress);
    }
```